pygmt/base_plotting.py

```diff
@@ -1922,3 +1922,248 @@ def update_pointers(data_pointers):
+            :gmt-docs:`supplements/geodesy/velo.html#vector-attributes`.
+
+        {B}
+
+        {CPT}
+
+        rescale : str
+            can be used to rescale the uncertainties of velocities
+            (``scaling='e'`` and ``scaling='r'``) and rotations
+            (``scaling='w'``). Can be combined with the confidence variable.
+
+        uncertainty_color : str
+            Sets the color or shade used for filling uncertainty wedges
+            (``scaling='w'``) or velocity error ellipses (``scaling='e'`` or
+            ``scaling='r'``). [If *uncertainty_color* is not specified, the
+            uncertainty regions will be transparent]. More details on
+            :gmt-docs:`cookbook/features.html#gfill-attrib`.
+
+        facecolor : str
+            Select color or pattern for filling of symbols or polygons
+            [Default is no fill]. More details on
+            :gmt-docs:cookbook/features.html#gfill-attrib`.
+
+        line: str
+            Draw lines. Ellipses and fault planes will have their outlines
+            drawn using current pen (see *pen*).
+
+        no_clip: str
+            Do NOT skip symbols that fall outside the frame boundary specified
+            by *region*. [Default plots symbols inside frame only].
+
+        pen : str
+            Set pen attributes for velocity arrows, ellipse circumference and
+            fault plane edges. [Defaults: width = default, color = black,
+            style = solid].
+
+        {U}
+        {V}
+        {XY}
+        {p}
+        {t}
+        """
+        kwargs = self._preprocess(**kwargs)
+
+        kind = data_kind(data)
+
+        extra_arrays = []
+        if "G" in kwargs and not isinstance(kwargs["G"], str):
+            if kind != "vectors":
+                raise GMTInvalidInput(
+                    "Can't use arrays for color if data is matrix or file."
+                )
+            extra_arrays.append(kwargs["G"])
+            del kwargs["G"]
+
+        if scaling is not None:
+            if not isinstance(scaling,str):
+                raise GMTInvalidInput(
+                    "Scaling is a requirred argument and has to be a string."
+                )
+            extra_arrays.append(scaling)
+
+        with Session() as lib:
+            # Choose how data will be passed in to the module
+            if kind == "file":
+                file_context = dummy_context(data)
+            elif kind == "matrix":
+                if pd.api.types.is_numeric_dtype(data):
+                    file_context = lib.virtualfile_from_matrix(data)
+                elif isinstance(data,np.ndarray):
+                    raise GMTInvalidInput(
+                        "Text columns are not supported with numpy array. "
+                        "They are only supported with file or "
+                        "pandas dataframe inputs."
+                    )
+                elif isinstance(data,pd.core.frame.DataFrame):
```

Comment:
This check might leave out other `dataframe` like objects (e.g. dask.dataframe), are you sure about this?

Reply:
hey @weiji14, I didnt realise they could be other type of `dataframe`. What about that adding that instead:
```python
type(data).__name__.lower().find('dataframe') != 1`
```
like in https://github.com/GenericMappingTools/pygmt/pull/525/commits/92f94f199cf86e314c6725aacd89c069d2289260

Reply:
Could we just have an `else` statement instead of an `elif` and `else`? This would simplify the code and I don't think the error will look too scary if someone enters a non-compatible data format (but happy to consider otherwise).